### ISSUE TITLE
Show box (/swarm) plots for categorical metadata plots, and scatter plots for numerical metadata plots

### ISSUE THREAD

**Issue by fedarko** [fedarko/rankratioviz#11]
e.g. box plots, like in https://github.com/knightlab-analyses/reference-frames/blob/master/ipynb/Figure3.ipynb

**Comment by fedarko** [comment]
This should be added after #31 is taken care of. The user might set the x-axis to a boolean (only 0 or 1) metadata column, for example, which would be better represented with a box plot than a scatterplot. Or the user might want to see a bar chart (e.g. if they select something like "age" as the x-axis).

We could try to autodetect the "optimal" chart type based on the metadata, but I think it's probably best (and most intuitive for the user) if we let the user control the plot design used (this would of course require various warnings if there are things like missing/NaN metadata fields, or type issues [i.e. strings]). Will probably require a decent amount of work, but it'll be worth it.

**Comment by fedarko** [comment]
From #31:

> So since it doesn't look like there's an easy way to make Vega* work with dynamic scale types, I think this is a solution:
> 
> - For the input sample metadata file, if it doesn't have a Q2 type row, just use `pandas.infer_objects()` to figure out each column's type. If it's a numeric type (int, float, ...) then record that column as numeric, and otherwise record that column as categorical. (This should roughly match how QIIME 2 infers metadata types -- see [the Q2 metadata docs](https://docs.qiime2.org/2019.1/tutorials/metadata/).)
> - If the sample metadata *has* a Q2 type row, use those types to determine numeric/categorical fields.
> - Create two sample plot JSON specs: one for a scatterplot with all the quantitative metadata fields, and one for a box / violin plot with all the categorical metadata fields. (Each sample should have a `rankratioviz_balance` and `Sample ID` metadata field, at least, so there should always be a sample plot JSON of each type created.)
> - The sample plot JSONs should only contain metadata information for their corresponding metadata field(s). Eventually, we can optimize things so that feature counts, feature column IDs, etc. within samples are stored in an external data array or something in `main.js` (so that they can be referenced by `change` operations for both sample plots).
> - Each sample plot JSON should have an x-axis signal for just their associated metadata fields. I guess we could have some rough UI element with "switch to quantitative fields" and "switch to categorical fields", which changes plot visibility? That'd also toggle signal visibility I guess. Eventually it'd be nice to have all metadata fields in one big dropdown but not sure of feasibility of that.

**Comment by fedarko** [comment]
Got a box plot mocked up in Vega-Lite! An important sticking point that was causing trouble: you need to make sure that the default rrv tooltip is removed (since it's done by sample ID, I guess?). Making the tooltip by `Station` (which I'm using as the x-axis in the boxplot) or just removing it entirely works perfectly. (V-L replaces the tooltip with a fancy generated one anyway showing some stats w/r/t the balance.) Also, the x-axis type needs to be nominal (or ordinal, I guess, but nominal is probs the better general option).

One other potential pain point: if any x-axis fields only have one sample belonging to them, they'll show up as a white line which is basically invisible on a white background. I think we can address this by adjusting the `median` tick property of the box plot (I guess we should just make the median line something that stands out -- maybe, like, gray or something).

**Comment by fedarko** [comment]
Per discussion with @cameronmartino, it'd be a good idea to use swarm plots (i.e. the individual sample points are overlaid on top of the box plot). See https://vega.github.io/vega/examples/beeswarm-plot/ for an example of doing this in Vega.
